src/test/java/hudson/plugins/git/extensions/impl/PruneStaleTagTest.java

```diff
@@ -0,0 +1,237 @@
+        remoteClient.tag(tagName, tagComment);
+        extension.decorateFetchCommand(scm, run, localClient, listener, null);
+        Assert.assertFalse("new tags should not be fetched", localClient.tagExists(tagName));
+    }
+
+    /*
+     * Local tag       | Remote tag
+     * -------------------------------
+     * exists - same   | exists - same
+     */
+    @Test
+    public void verify_that_local_tag_is_not_pruned_when_exist_on_remote() throws Exception {
+        File remoteRepo = fileRule.newFolder("remote");
+
+        // create a remote repository with one tag
+        GitClient remoteClient = initRepository(listener, remoteRepo);
+        String tagName = "tag";
+        String tagComment = "tag comment";
+        remoteClient.tag(tagName, tagComment);
+
+        // clone remote repository to workspace
+        GitClient localClient =  cloneRepository(remoteRepo);
+
+        GitSCM scm = new GitSCM(localClient.getRemoteUrl("origin"));
+        PruneStaleTag extension = new PruneStaleTag();
+
+        extension.decorateFetchCommand(scm, run, localClient, listener, null);
+        Assert.assertTrue("local tags must not be pruned when exists on remote", localClient.tagExists(tagName));
+    }
+
+    /*
+     * Local tag       | Remote tag
+     * -------------------------------
+     * exists          | not exists
+     */
+    @Test
+    public void verify_that_local_tag_is_pruned_when_not_exist_on_remote() throws Exception {
+        File remoteRepo = fileRule.newFolder("remote");
+
+        // create a remote repository with one tag
+        GitClient remoteClient = initRepository(listener, remoteRepo);
+        String tagName = "tag";
+        String tagComment = "tag comment";
+        remoteClient.tag(tagName, tagComment);
+
+        // clone remote repository to workspace
+        GitClient localClient =  cloneRepository(remoteRepo);
+
+        GitSCM scm = new GitSCM(localClient.getRemoteUrl("origin"));
+        PruneStaleTag extension = new PruneStaleTag();
+
+        // remove tag on remote, tag remains on local cloned repository
+        remoteClient.deleteTag(tagName);
+        extension.decorateFetchCommand(scm, run, localClient, listener, null);
+        Assert.assertFalse("local tag has not been pruned", localClient.tagExists(tagName));
+    }
+
+    @Test
+    public void verify_fetch_do_not_prune_local_branches() throws Exception {
+        File remoteRepo = fileRule.newFolder("remote");
+
+        // create a remote repository with one tag
+        initRepository(listener, remoteRepo);
+
+        // clone remote repository to workspace
+        File localRepo = fileRule.newFolder("local");
+        GitClient localClient = new TestJGitAPIImpl(localRepo, listener);
+        String gitURL = remoteRepo.toURI().toString();
+        localClient.clone(gitURL, "origin", false, "+refs/heads/*:refs/remotes/origin/*");
+        localClient.checkoutBranch("master", "refs/remotes/origin/master");
```

Comment:
I might be missing something, but cannot be done using the `cloneRepository`?
```suggestion
        // clone remote repository to workspace
        GitClient localClient =  cloneRepository(remoteRepo);
```